verilog/tools/kythe/indexing_facts_tree_extractor.h

```diff
@@ -224,6 +224,9 @@ class IndexingFactsTreeExtractor : public verible::TreeContextVisitor {
   // e.g counter #(.N(r)) extracts "N".
   void ExtractParamByName(const verible::SyntaxTreeNode& param_by_name);
 
+  // Creates new scope with unique id for if conditions and loops.
+  void ExtractTemporaryScope(const verible::SyntaxTreeNode& node);
```

Comment:
Design: this is not really a temporary scope, it needs to persist semantically, even if it is not named.  It should not be discarded, which is what "Temporary" is suggesting.

Reply:
+1

Unnamed or Loop scope sounds better. And it should persist.